File: IPC/pract/pipe/pipe2.c
#include<stdio.h>
int main()
{
	int p[2];
	pipe(p);
	if(fork()==0)
	{
		int x=1;
		for(x=1;x<=10;x++)
		{
			puts("writing");
			sleep(1);
			write(p[1],&x,4);
		}

	}
	else
	{
		int y;
		sleep(10);
		do{
			read(p[0],&y,4);
			printf("%d\n",y);
			sleep(1);
		}while(y!=10);

	}


}

File: python/fork/pipe2.py
import os, sys

#print "The child will write text to a pipe and "
#print "the parent will read the text written by child..."

# file descriptors r, w for reading and writing
re, wr = os.pipe() 
v=3
processid = os.fork()
if processid:
    # This is the parent process 
    # Closes file descriptor w
	#v+=10
	#print "v:",v
	os.close(wr)
	r = os.fdopen(re)
	print "Parent reading"
	str = r.read()
	print "text =", str   
	sys.exit(0)
else:
    # This is the child process
	v+=10
	os.close(re)
	w = os.fdopen(wr, 'w')
	print "Child writing"
	#w.write("Text written by child...")
	w.write(str(v))
	w.close()
	print "Child closing"
	sys.exit(0)
